sklearn/preprocessing/_data.py

```diff
@@ -514,18 +520,20 @@ def transform(self, X):
         """
         check_is_fitted(self)
 
+        xp, _ = get_namespace(X)
+
         X = self._validate_data(
             X,
             copy=self.copy,
-            dtype=FLOAT_DTYPES,
+            dtype=(xp.float64, xp.float32, xp.float16),
```

Comment:
If there is a another Array API implementation that does not have `float16`, I do not want our Array API wrapper to break. The Array API spec only enforces that `float32` and `float64` exists.

```python
def _supported_float_dtypes(xp):
    if hasattr(xp, "float16"):
        return {xp.float16, xp.float32, xp.float64}
    else:
        return {xp.float32, xp.float64}
```

Reply:
This is a smart idea. Done.